test/support/src/helpers.cc

```diff
@@ -832,6 +827,75 @@ void create_ctx_and_vfs(
   tiledb_config_free(&config);
 }
 
+void get_supported_fs(
+    bool* s3_supported,
+    bool* hdfs_supported,
+    bool* azure_supported,
+    bool* gcs_supported) {
+  tiledb_ctx_t* ctx = nullptr;
+  REQUIRE(tiledb_ctx_alloc(nullptr, &ctx) == TILEDB_OK);
+
+  if constexpr (tiledb::sm::filesystem::s3_enabled) {
+    *s3_supported = true;
+  }
+
+  if constexpr (tiledb::sm::filesystem::hdfs_enabled) {
+    *hdfs_supported = true;
+  }
+
+  if constexpr (tiledb::sm::filesystem::azure_enabled) {
+    *azure_supported = true;
+  }
+
+  if constexpr (tiledb::sm::filesystem::gcs_enabled) {
+    *gcs_supported = true;
+  }
+
+  // Override VFS support if the user used the '--vfs' command line argument.
+  if (!g_vfs.empty()) {
```

Comment:
I'd put the previous set of statements the `if` clause and this one the `else` clause.